packages/mock-block-dock/src/use-mock-block-props.ts

```diff
@@ -24,79 +25,71 @@ import { MockData, useMockDatastore } from "./use-mock-datastore";
  * @param [initialLinkedAggregations] - The linkedAggregation DEFINITIONS to include in the data store (results will be resolved automatically)
  */
 export const useMockBlockProps = ({
-  blockProperties,
+  blockEntity,
   blockSchema,
   initialEntities,
   initialEntityTypes,
   initialLinks,
   initialLinkedAggregations,
 }: {
-  blockProperties: BlockProtocolEntity;
-  blockSchema?: Partial<BlockProtocolEntityType>;
-  initialEntities?: BlockProtocolEntity[];
-  initialEntityTypes?: BlockProtocolEntityType[];
-  initialLinks?: BlockProtocolLink[];
-  initialLinkedAggregations?: BlockProtocolLinkedAggregationDefinition[];
+  blockEntity: Entity;
+  blockSchema?: Partial<EntityType>;
+  initialEntities?: Entity[];
+  initialEntityTypes?: EntityType[];
+  initialLinks?: Link[];
+  initialLinkedAggregations?: LinkedAggregationDefinition[];
 }): {
-  blockProperties: BlockProtocolEntity;
-  blockProtocolFunctions: Required<BlockProtocolFunctions>;
-  entityTypes: BlockProtocolProps["entityTypes"];
-  linkedAggregations: BlockProtocolProps["linkedAggregations"];
-  linkedEntities: BlockProtocolProps["linkedEntities"];
-  linkGroups: BlockProtocolProps["linkGroups"];
+  blockEntity: Entity;
+  blockGraph: BlockGraph;
+  entityTypes: EntityType[];
+  graphServiceCallbacks: Required<EmbedderGraphMessageCallbacks>;
+  linkedAggregations: LinkedAggregation[];
 } => {
   const { initialBlockEntity, mockData } = useMemo((): {
-    initialBlockEntity: BlockProtocolEntity;
+    initialBlockEntity: Entity;
     mockData: MockData;
   } => {
-    const blockEntityType: BlockProtocolEntityType = {
-      entityTypeId: "blockType1",
-      title: "BlockType",
-      type: "object",
-      $schema: "https://json-schema.org/draft/2019-09/schema",
-      $id: "http://localhost/blockType1",
-      ...(blockSchema ?? {}),
+    const entityTypeId = blockEntity?.entityTypeId ?? "block-type-1";
+
+    const blockEntityType: EntityType = {
+      entityTypeId,
+      schema: {
+        title: "BlockType",
+        type: "object",
+        $schema: "https://json-schema.org/draft/2019-09/schema",
+        $id: "http://localhost/blockType1",
+        ...(blockSchema ?? {}),
+      },
     };
 
-    const accountId = blockProperties.accountId ?? "accountId";
-
-    const blockEntity: BlockProtocolEntity = {
-      accountId,
+    const newBlockEntity: Entity = {
       entityId: "block1",
+      entityTypeId,
+      properties: {},
```

Comment:
I'm wondering if empty properties as the initial object is a good idea? It would be invalid properties for any schema that has required fields, and the blocks will have to check if `properties` is an empty object before using it as props. I think `undefined` would make it clear that no props have been set yet.

Reply:
For context, in the calculation block, i assume the props can always be deserialized, which fails on initial load
![image](https://user-images.githubusercontent.com/6988668/172852696-a55122e3-c529-408c-8fe3-ad686d53aa4c.png)

Reply:
We discussed and landing on making the `properties` field optional, as the spec says, which I've done in [make entityTypeId and properties optional in graph Entity type](https://github.com/blockprotocol/blockprotocol/pull/326/commits/d7b90e57c97e05134cf7fd922608897659400e9e) - along with making `entityTypeId` optional, as per the spec.

We decided not to change the 'empty properties' object, since blocks should be able to handle an empty object if they are capable of being loaded with no props, and if they express `required` properties in their schema, embedding applications should not be loading them at all without the necessary properties present in the object.